app/workers/delete_object_hierarchy_worker.rb

```diff
@@ -21,7 +21,7 @@ class DeleteObjectHierarchyWorker < ApplicationJob
     info "Finished #{job.class}#perform with the hierarchy of workers: #{caller_worker_hierarchy}"
   end
 
-  def perform(_object, _caller_worker_hierarchy = [], _destroy_method = 'destroy')
+  def perform(_object, _caller_worker_hierarchy = [], _background_destroy_method = 'destroy')
```

Comment:
Personally I prefer in this case to keep it like `(_object, _caller_worker_hierarchy = [], _background_destroy_method = 'destroy')`  instead of `perform(*)` because:
1. It has default values by doing it this way.
2. All of them are actually used, but saved in the `before_perform`, so this way we enforce to send exactly 3 params while the other way we accept anything and it is prone to errors. Besides, putting them names makes more clear what we expect and what each param means.